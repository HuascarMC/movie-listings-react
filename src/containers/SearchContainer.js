import React from 'react';
import axios from 'axios'
import GenreListContainer from './GenreListContainer'

class SearchContainer extends React.Component {

  render() {
    return(
      <div className="search-wrapper">
        <GenreListContainer />
      </div>
    );
  }
}

export default SearchContainer;
