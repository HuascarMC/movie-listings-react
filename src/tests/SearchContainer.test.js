import React from 'react';
import SearchContainer from '../containers/SearchContainer'
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

it('should render without crashing', () => {
  shallow(<SearchContainer />);
});

it('should render InputContainer', () => {
  const wrapper = mount(<SearchContainer />);
  const container = wrapper.find('InputContainer');
  expect(container.length).toEqual(1);
});

it('should render InputContainer with collapse prop', () => {
  const wrapper = mount(<SearchContainer collapse=""/>);
  const container = wrapper.find('InputContainer');
  expect(container.prop("collapse")).toEqual("");
});

it('should render InputContainer with updateSearch prop', () => {
  const wrapper = mount(<SearchContainer updateSearch=""/>);
  const container = wrapper.find('InputContainer');
  expect(container.prop("updateSearch")).toEqual("");
});

xit('should render GenreListContainer', () => {
  const wrapper = mount(<SearchContainer />);
  const container = wrapper.find('GenreListContainer');
  expect(container.length).toEqual(1);
});

it('should match the snapshot', () => {
  const component = renderer.create(<SearchContainer />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
