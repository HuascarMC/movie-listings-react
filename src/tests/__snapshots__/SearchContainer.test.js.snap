// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  className="search-wrapper"
>
  <div
    className="genre-container"
  >
    <div
      className="genre-component"
    >
      <p>
        comedy
      </p>
    </div>
    <div
      className="genre-component"
    >
      <p>
        action
      </p>
    </div>
    <div
      className="genre-component"
    >
      <p>
        indie
      </p>
    </div>
  </div>
</div>
`;
